10.0.2: 'No GlobalOverrides context is active' warning from get_version

All of our [builds are now raising this warning](https://github.com/matplotlib/matplotlib/actions/runs/23568084983/job/68624305839?pr=31198#step:13:940) in every test (which is turned into an error on pytest):
```pytb
lib/matplotlib/__init__.py:227: in _get_version
    return setuptools_scm.get_version(
/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/vcs_versioning/_get_version_impl.py:246: in get_version
    maybe_version = _get_version(config, force_write_version_files=True)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/vcs_versioning/_get_version_impl.py:101: in _get_version
    parsed_version = parse_version(config)
                     ^^^^^^^^^^^^^^^^^^^^^
/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/vcs_versioning/_get_version_impl.py:59: in parse_version
    or parse_scm_version(config)
       ^^^^^^^^^^^^^^^^^^^^^^^^^
/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/vcs_versioning/_get_version_impl.py:37: in parse_scm_version
    return _entrypoints.version_from_entrypoint(
/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/vcs_versioning/_entrypoints.py:29: in version_from_entrypoint
    maybe_version: ScmVersion | None = fn(root, config=config)
                                       ^^^^^^^^^^^^^^^^^^^^^^^
/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/vcs_versioning/_backends/_git.py:308: in parse
    _require_command("git")
/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/vcs_versioning/_run_cmd.py:210: in require_command
    if not has_command(name, warn=False):
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/vcs_versioning/_run_cmd.py:187: in has_command
    p = run([name, *args], cwd=".")
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^
/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/vcs_versioning/_run_cmd.py:146: in run
    timeout = _get_timeout(os.environ)
              ^^^^^^^^^^^^^^^^^^^^^^^^
/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/vcs_versioning/_run_cmd.py:23: in _get_timeout
    return get_subprocess_timeout()
           ^^^^^^^^^^^^^^^^^^^^^^^^
/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/vcs_versioning/overrides.py:609: in get_subprocess_timeout
    return get_active_overrides().subprocess_timeout
           ^^^^^^^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def get_active_overrides() -> GlobalOverrides:
        """Get the currently active GlobalOverrides instance.
    
        If no context is active, creates one from the current environment
        using SETUPTOOLS_SCM prefix for legacy compatibility.
    
        Note: The auto-created instance reads from os.environ at call time,
        so it will pick up environment changes (e.g., from pytest monkeypatch).
    
        Returns:
            GlobalOverrides instance
        """
        global _auto_create_warning_issued
    
        overrides = _active_overrides.get()
        if overrides is None:
            # Auto-create context from environment for backwards compatibility
            # Note: We create a fresh instance each time to pick up env changes
            if not _auto_create_warning_issued:
>               warnings.warn(
                    "No GlobalOverrides context is active. "
                    "Auto-creating one with SETUPTOOLS_SCM prefix for backwards compatibility. "
                    "Consider using 'with GlobalOverrides.from_env(\"YOUR_TOOL\"):' explicitly.",
                    UserWarning,
                    stacklevel=2,
                )
E               UserWarning: No GlobalOverrides context is active. Auto-creating one with SETUPTOOLS_SCM prefix for backwards compatibility. Consider using 'with GlobalOverrides.from_env("YOUR_TOOL"):' explicitly.

/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/vcs_versioning/overrides.py:578: UserWarning
```

I can configure pytest to ignore the warning, but I have no idea why it's generated in the first place. If it matters, this is an editable install of the project, and [the `get_version` call](https://github.com/matplotlib/matplotlib/blob/a85564ead5868dcc2244e42b75b7dbe0d13fb8b7/lib/matplotlib/__init__.py#L227-L233) is:
```python
            return setuptools_scm.get_version(
                root=root,
                dist_name="matplotlib",
                version_scheme="release-branch-semver",
                local_scheme="node-and-date",
                fallback_version=_version.version,
            )
```

--- a/setuptools-scm/changelog.d/1314.bugfix.md
+++ b/setuptools-scm/changelog.d/1314.bugfix.md
@@ -1,0 +1,1 @@
+Enter ``GlobalOverrides`` for ``SETUPTOOLS_SCM`` when using ``setuptools_scm.get_version`` / ``_get_version``, avoiding implicit context warnings for direct API callers.

--- a/setuptools-scm/src/setuptools_scm/__init__.py
+++ b/setuptools-scm/src/setuptools_scm/__init__.py
@@ -12,8 +12,9 @@ from vcs_versioning import Version
 from vcs_versioning._config import DEFAULT_LOCAL_SCHEME
 from vcs_versioning._config import DEFAULT_VERSION_SCHEME
 from vcs_versioning._dump_version import dump_version  # soft deprecated
-from vcs_versioning._get_version_impl import _get_version
-from vcs_versioning._get_version_impl import get_version  # soft deprecated
+
+from ._get_version import _get_version
+from ._get_version import get_version  # soft deprecated
 
 # Public API
 __all__ = [

--- a/setuptools-scm/src/setuptools_scm/_get_version.py
+++ b/setuptools-scm/src/setuptools_scm/_get_version.py
@@ -1,0 +1,76 @@
+"""setuptools-scm wrappers for vcs_versioning get_version APIs.
+
+``get_version`` and ``_get_version`` enter :class:`~vcs_versioning.overrides.GlobalOverrides`
+for the ``SETUPTOOLS_SCM`` tool prefix so callers (e.g. ``setup.py`` / library code) do not
+trigger implicit context auto-creation warnings from vcs_versioning.
+"""
+
+from __future__ import annotations
+
+import logging
+
+from re import Pattern
+from typing import Any
+
+from vcs_versioning import _config
+from vcs_versioning import _types as _t
+from vcs_versioning._config import Configuration
+from vcs_versioning._get_version_impl import _get_version as _get_version_core
+from vcs_versioning._get_version_impl import get_version as _get_version_public
+from vcs_versioning.overrides import ensure_context
+
+_setuptools_scm_logger = logging.getLogger("setuptools_scm")
+
+
+def _get_version(
+    config: Configuration, force_write_version_files: bool | None = None
+) -> str | None:
+    with ensure_context("SETUPTOOLS_SCM", additional_loggers=_setuptools_scm_logger):
+        return _get_version_core(
+            config, force_write_version_files=force_write_version_files
+        )
+
+
+def get_version(
+    root: _t.PathT = ".",
+    version_scheme: _t.VERSION_SCHEME = _config.DEFAULT_VERSION_SCHEME,
+    local_scheme: _t.VERSION_SCHEME = _config.DEFAULT_LOCAL_SCHEME,
+    write_to: _t.PathT | None = None,
+    write_to_template: str | None = None,
+    version_file: _t.PathT | None = None,
+    version_file_template: str | None = None,
+    relative_to: _t.PathT | None = None,
+    tag_regex: str | Pattern[str] = _config.DEFAULT_TAG_REGEX,
+    parentdir_prefix_version: str | None = None,
+    fallback_version: str | None = None,
+    fallback_root: _t.PathT = ".",
+    parse: Any | None = None,
+    git_describe_command: _t.CMD_TYPE | None = None,
+    dist_name: str | None = None,
+    version_cls: Any | None = None,
+    normalize: bool = True,
+    search_parent_directories: bool = False,
+    scm: dict[str, Any] | None = None,
+) -> str:
+    with ensure_context("SETUPTOOLS_SCM", additional_loggers=_setuptools_scm_logger):
+        return _get_version_public(
+            root=root,
+            version_scheme=version_scheme,
+            local_scheme=local_scheme,
+            write_to=write_to,
+            write_to_template=write_to_template,
+            version_file=version_file,
+            version_file_template=version_file_template,
+            relative_to=relative_to,
+            tag_regex=tag_regex,
+            parentdir_prefix_version=parentdir_prefix_version,
+            fallback_version=fallback_version,
+            fallback_root=fallback_root,
+            parse=parse,
+            git_describe_command=git_describe_command,
+            dist_name=dist_name,
+            version_cls=version_cls,
+            normalize=normalize,
+            search_parent_directories=search_parent_directories,
+            scm=scm,
+        )

--- a/setuptools-scm/testing_scm/test_integration.py
+++ b/setuptools-scm/testing_scm/test_integration.py
@@ -58,6 +58,39 @@ def wd(wd: WorkDir, monkeypatch: pytest.MonkeyPatch) -> WorkDir:
     return wd.setup_git(monkeypatch)
 
 
+@pytest.mark.issue("1314")
+def test_get_version_no_implicit_global_overrides_warning_subprocess(
+    wd: WorkDir,
+) -> None:
+    """setuptools_scm.get_version must not rely on implicit GlobalOverrides auto-create.
+
+    Pytest's vcs_versioning plugin installs a global ``GlobalOverrides`` context, so
+    this check runs in a subprocess without that fixture (see GitHub issue #1314).
+    """
+    root = os.fspath(wd.cwd)
+    script = textwrap.dedent(
+        f"""
+        import warnings
+
+        warnings.filterwarnings(
+            "error",
+            message="No GlobalOverrides context is active",
+            category=UserWarning,
+        )
+        import setuptools_scm
+
+        setuptools_scm.get_version(root={root!r})
+        """
+    )
+    proc = subprocess.run(
+        [sys.executable, "-c", script],
+        capture_output=True,
+        text=True,
+        check=False,
+    )
+    assert proc.returncode == 0, (proc.stdout, proc.stderr)
+
+
 def test_pyproject_support(tmp_path: Path, monkeypatch: pytest.MonkeyPatch) -> None:
     monkeypatch.delenv("SETUPTOOLS_SCM_DEBUG")
     pkg = tmp_path / "package"

